pandas/_libs/tslibs/timezones.pyx

```diff
@@ -2,6 +2,7 @@ from datetime import (
     timedelta,
     timezone,
 )
+from zoneinfo import ZoneInfo
```

Comment:
This is an unconditional import — do you not require support for Python < 3.8?

If so, you should presumably have some more complicated logic here and in the `is_zoneinfo` logic.

Something like this works well for "is this an instance of X" without forcing the user to import the module that contains X:

https://github.com/pganssle/pytz-deprecation-shim/blob/bf0174adfce698e280c623d17fe8e82961de0c48/src/pytz_deprecation_shim/helpers.py#L11-L30

https://github.com/pganssle/pytz-deprecation-shim/blob/bf0174adfce698e280c623d17fe8e82961de0c48/src/pytz_deprecation_shim/_common.py#L6-L13

Since obviously some object isn't going to be an instance of `zoneinfo.ZoneInfo` if the interpreter has never imported the `zoneinfo` module.

Reply:
> This is an unconditional import — do you not require support for Python < 3.8?

I think our minimum version is 3.8, but I'm on 3.9 locally and forgot when writing this branch.  Will change to try/except.

Reply:
no leave this

we support >= 3.8

Reply:
zoneinfo isnt stdlib until 3.9, so importing unconditionally would require making backports.zoneinfo a hard dependency (and having that in the CI dep file is causing problems AFAICT)

Reply:
Even with `backports.zoneinfo` it needs to be conditional since they're in different namespaces, plus I really like the approach I took in the `pytz` deprecation shims for checking if a time zone is a `pytz` zone, where my checks never actually import `pytz` unless it's already been imported (since obviously if it's never been imported then I know the object isn't a `pytz` type). That may be overkill, but it's also not that hard to implement, so :balance_scale:

Reply:
I'll need to take a closer look at the pytz deprecation shim.  We're getting rid of a lot of pytz usage in 2.0 (#34916) and these days I keep finding ways we could simplify the code even more if we dropped pytz support altogether.